### PR TITLE
nixos/security.acme: remove `with lib;`

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #208242

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilazy** [comment]
@ofborg test acme

**Comment by m1cr0man** [comment]
Correct me if I'm wrong but `builtins.lib` is not a thing? It seems some mass replace was done across the file.

**Comment by emilazy** [comment]
Yeah, this needs fixing.

**Comment by Stunkymonkey** [comment]
@emilazy & @m1cr0man should be fixed.

**Comment by Stunkymonkey** [comment]
@ofborg test acme

File: nixos/modules/security/acme/default.nix [hunk 1]
```diff
@@ -196,30 +195,30 @@ let
     # FIXME when mkChangedOptionModule supports submodules, change to that.
     # This is a workaround
     extraDomains = data.extraDomainNames ++ (
-      optionals
-      (data.extraDomains != "_mkMergedOptionModule")
+      lib.optionals
+      (data.extraDomains != "_lib.mkMergedOptionModule")
```
Comment (emilazy) [inline nixos/modules/security/acme/default.nix:199]: I very much doubt this is correct. If the test passes then we should probably fix it to exercise this code path.

**Comment by m1cr0man** [comment]
Question (non blockers just my own curiosity): what's the value of removing "with lib" entirely as per the parent ticket? I also wonder if a "let inherit (lib) ...” pattern is worth considering to preserve the blame on the lines/keep common values in scope?

File: nixos/modules/security/acme/default.nix (hunk 1 above)
Comment (m1cr0man) [reply nixos/modules/security/acme/default.nix:199]: I had a pending comment for the same 👍

**Comment by emilazy** [comment]
`with` disables static scope checking for its whole body, and has very unintuitive behaviour around variable shadowing. It’s controversial in general but there’s more‐or‐less consensus to avoid whole file bodies being inside a `with` at this point. (And insofar as my vote as someone who hasn’t touched the ACME code in a few years counts, I’m definitely in favour of dropping it.)

I am personally agnostic to explicit `lib` vs. `let inherit`. I write the former, but the latter is fine too. Opinions seem generally divided on the matter.

**Comment by m1cr0man** [comment]
Thanks for the explanation 🙂 this change makes a lot of sense now.

**Comment by Stunkymonkey** [comment]
@ofborg test acme

**Review by philiptaron — APPROVED** [review]
`nix-build -A nixosTests.acme` passes.

**Review by m1cr0man — APPROVED** [review]
Looks good 🙂

**Pull request merged**